### PR TITLE
ARRISEOS-46916: [WPE 2.38] matchMedia("(dynamic-range: high)") always…

### PR DESCRIPTION
**Pull request by amol-virnodkar-infosys**
… returns matches=false

### CODE REVIEW

**Comment by amol-virnodkar-infosys** [comment]
Addressed all the review comments in the PR: https://github.com/LibertyGlobal/rdkservices/pull/414 provided by Mikolaj in this new PR.

File: WebKitBrowser/WebKitImplementation.cpp [hunk 1]
```diff
@@ -2937,6 +2948,57 @@ static GSourceFuncs _handlerIntervention =
...
+            {
+                TRACE_L1("Error initializing IARM.. error code : %d\n",err);
+                return false;
+            }
+
+            err = IARM_Bus_Connect();
+            if(IARM_RESULT_SUCCESS != err)
+            {
+                TRACE_L1("Error connecting to IARM.. error code : %d\n",err);
+                return false;
```
Comment (marcin-mielczarczyk-red) [inline WebKitBrowser/WebKitImplementation.cpp]: IARM_Bus_Term() should be called here.

File: WebKitBrowser/WebKitImplementation.cpp [hunk 2]
```diff
@@ -2937,6 +2948,57 @@ static GSourceFuncs _handlerIntervention =
...
+            {
+                TRACE_L1("STB and TV supports HDR : returning true\r\n");
+                return true;
+            }
+
+            device::Manager::DeInitialize();
+            IARM_Bus_Disconnect();
+            IARM_Bus_Term();
+            TRACE_L1("GetHDRCapabilities : returning false\r\n");
+            return false;
```
Comment (marcin-mielczarczyk-red) [inline WebKitBrowser/WebKitImplementation.cpp]: Shouldn't we deinitialize IARM even when HDR is supported?
Is it needed later?
I would also prefer to have a single print for false and true for easier parsing:

TRACE_L1("GetHDRCapabilities : returning %s\r\n", ret ? "true" : "false");

**Review by marcin-mielczarczyk-red — CHANGES_REQUESTED** [review]

File: WebKitBrowser/WebKitImplementation.cpp (hunk 1 above)
Comment (amol-virnodkar-infosys) [reply WebKitBrowser/WebKitImplementation.cpp]: Addressed the review comment.

File: WebKitBrowser/WebKitImplementation.cpp (hunk 2 above)
Comment (amol-virnodkar-infosys) [reply WebKitBrowser/WebKitImplementation.cpp]: Addressed the review comment.

File: WebKitBrowser/WebKitImplementation.cpp [hunk 3]
```diff
@@ -2937,6 +2948,59 @@ static GSourceFuncs _handlerIntervention =
...
+            if(vPort.isDisplayConnected())
+            {
+                vPort.getTVHDRCapabilities(&tvCaps);
+            }
+
+            TRACE_L1("TV HDRCapabilities - [%d]\r\n", tvCaps);
+
+            if(stbCaps != 0 && tvCaps != 0)
+            {
+                TRACE_L1("STB and TV supports HDR : returning true\r\n");
```
Comment (marcin-mielczarczyk-red) [inline WebKitBrowser/WebKitImplementation.cpp]: I think we don't need this print as true is returned only in this if statement.

Comment (amol-virnodkar-infosys) [reply WebKitBrowser/WebKitImplementation.cpp]: Addressed the review comments.

**Review by marcin-mielczarczyk-red — APPROVED** [review]

**Review by suresh-khurdiya-infosys — APPROVED** [review]

**Pull request merged**